contract/create_user_request.go

```diff
@@ -0,0 +1,42 @@
+package contract
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+)
+
+type CreateUserRequest struct {
+	Name     string `json:"name"`
+	Email    string `json:"email"`
+	Password string `json:"password"`
+	IsAdmin  bool   `json:"is_admin"`
+}
+
+var emailRegex = regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+\\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
+
+func (req CreateUserRequest) Validate() error {
+	if strings.TrimSpace(req.Name) == "" {
+		return errors.New("Name is required")
+	}
+
+	email := req.Email
+
+	valid := emailRegex.MatchString(email)
+
+	if valid == false {
+
+		return errors.New("invalid email")
+
+	}
+
+	if strings.TrimSpace(req.Email) == "" {
```

Comment:
will this code be reachable in any case?

Reply:
yes I think I tested it.

Reply:
if the email is empty then that will be returned from above line where we are validating the email by regular expression